File: orcid_record.ts
import { verifyIdentifier } from "./utility.ts";
import { normalizeORCID, validateORCID } from "./orcid.ts";

/**
 * Verifies an ORCID using an external database, orcid.org
 * @param orcid - The identifier string to verify.
 * @returns Promise<boolean> - True if the ORCID exists, otherwise false.
 */
export async function verifyORCID(orcid: string): Promise<boolean> {
  const normalizedORCID = normalizeORCID(orcid);
  return await verifyIdentifier(
    orcid,
    `https://orcid.org/${encodeURIComponent(normalizedORCID)}`,
    validateORCID,
  );
}

File: orcid_record.go
package metadatatools

import "net/url"

// VerifyORCID verifies an ORCID iD exists via orcid.org. The identifier
// is normalized before verifying.
func VerifyORCID(orcid string) bool {
	normalized := NormalizeORCID(orcid)
	return VerifyIdentifier(orcid, "https://orcid.org/"+url.QueryEscape(normalized), ValidateORCID)
}
